### ISSUE TITLE
httpd_service actions :create and :delete do not de/provision systemd units

### ISSUE THREAD

**Issue by karmix** [chef-cookbooks/httpd#117]
### Cookbook version
0.4.5

### Chef-client version
13.3.42

### Platform Details
platforms:
  - name: ubuntu-16.04
  - name: centos-7.3

### Scenario:
We are unable to start the httpd service after configuring the httpd_service resource using "action :create" because httpd_service only defines the systemd unit in the :start action.

### Steps to Reproduce:
1. `chef generate cookbook httpd-bug`
2. add `depends 'httpd'` to metadata.rb
3. add the following three lines to recipes/default.rb
```
apt_update 'update'
httpd_service('proxy') { action :create }
service('httpd-proxy') { action :start }
```
4. `kitchen converge`

### Expected Result:
The httpd_service resource with only action :create should install and configure (but not start) all of the components of the Apache httpd service being defined.

### Actual Result:
The httpd_service resource installs and configures the Apache httpd application, but does not define the httpd-proxy service.

           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           Expected process to exit with [0], but received '5'
           ---- Begin output of /bin/systemctl --system start httpd-proxy ----
           STDOUT: 
           STDERR: Failed to start httpd-proxy.service: Unit not found.
           ---- End output of /bin/systemctl --system start httpd-proxy ----
           Ran /bin/systemctl --system start httpd-proxy returned 5

**Comment by karmix** [comment]
I note in #118 that the `:delete` action is similarly not removing the systemd units.

**Comment by tas50** [comment]
It may be helpful to migrate this cookbook over to systemd_unit to avoid some of the systemd pitfalls

**Comment by karmix** [comment]
Closing; resolved by #118.
